staff/acct/sorry/sorry

```diff
@@ -1,156 +1,183 @@
+
+
+def get_user_real_mail(username):
+    email = str(subprocess.run([shutil.which('ldapsearch'),
+                                'uid={0}'.format(username), 'mail'], stdout=subprocess.PIPE).stdout)
+    email = email[email.find('\\nmail') + 8:]
+    email = email[:email.find('\\n\\n')]
+    return email
+
+
+def log(root_staffer, sorried_user):
+    with open('/opt/acct/sorry.log', 'a') as log_file:
+        # Date style matters...?
+        log_file.write('{0} - {1} {2}'.format(str(datetime.datetime.now()), root_staffer, sorried_user))
+
+
+def send_log_mail(root_staffer, sorried_user, reason):
+    email_from = 'Open Computing Facility <sorry_log@ocf.berkeley.edu>'
+    email_subject = '[OCF] Account disabled'
+    email_body = '''
+This is an automated email sent by sorry script. This is the log of a recent sorry operation:
+Account "{0}" has been disabled by "{1}" for reason "{2}" on "{3}".
+'''.format(sorried_user, root_staffer, reason, str(datetime.datetime.now()))
+    ocflib.misc.mail.send_mail('root@ocf.berkeley.edu', email_subject, email_body, cc=None, sender=email_from)
+
+
+def send_sorry_mail(sorried_user_email, username, reason):
+    email_from = 'Open Computing Facility <help@ocf.berkeley.edu>'
+    email_subject = '[OCF] Account disabled'
+    email_body = '''
 Hello,
 
 Your OCF account has been disabled.
 
-Your account name is: ${sorriedUser}
+Your account name is: {0}
 
 The reason your account has been disabled:
-$(cat "$sorryFile")
+{1}
 
 Feel free to reply to this message.
 Please do not share your password with anyone or over email.
 
-$(python3 -c 'import ocflib.misc.mail as mail; print(mail.MAIL_SIGNATURE)')
-EOF
+{2}
+    '''.format(username, reason, ocflib.misc.mail.MAIL_SIGNATURE)
+    ocflib.misc.mail.send_mail(sorried_user_email, email_subject, email_body, cc=None, sender=email_from)
+
+
+def main():
+    sorry_description = 'Sorry a user. \nStandard sorry reasons are: \n' + '\n'.join(
+        [f for f in os.listdir(reason_path) if
+         os.path.isfile(os.path.join(reason_path, f))]) + '\nBut you can also pass in your own file.'
+    parser = argparse.ArgumentParser(usage='%(prog)s sorriedUser sorryReason',
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description=sorry_description)
+    parser.add_argument('sorriedUser', metavar='sorriedUser', help='Username of sorried user')
+    parser.add_argument('sorryReason', metavar='sorriedReason', help='Reason for the user to be sorried')
+
+    if (len(sys.argv) != 3):
+        parser.print_help()
+        return 1
+
+    if os.getuid() != 0:
+        raise PermissionError('You must be root to run this.')
+
+    args = parser.parse_args()
+    file_path = find_sorry_file(args.sorryReason)
+    print('Sorry file {}.'.format(file_path))
+
+    sorry_user(args.sorriedUser, file_path)
+    print('NOW RUN note -u {0} reason. NOW'.format(args.sorriedUser))
```

Comment:
It would be good to do this automatically. We should keep in mind that `note` does not have a way to read the reason from stdin by default, and the `reason` here should usually have more detail than the reason given in the sorryfile.